lib/companies/job_checker.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Companies.JobChecker do
+  alias Companies.Jobs
+
+  def check_job(job) do
```

Comment:
If possible, I'd like to avoid pipelines in situations like this. What do you think about capturing `url`? I think we can also eliminate the duplicate `nil` paths with a `_` match:

```suggestion
def check_job(%{url: url} = job) do
  with {:ok, %{status_code: 404}} <- HTTPoison.head(url) do
    Jobs.delete(job)
  else
    _ -> nil
  end
end
```

```suggestion
def check_job(%{url: url} = job) do
  case HTTPoison.head(url) do
    {:ok, %{status_code: 404}} -> Jobs.delete(job)
    _ -> nil
  end
end
```